dataland-frontend/src/components/general/SelectReportingPeriodDialog.vue

```diff
@@ -40,9 +40,10 @@ export default defineComponent({
      */
     reportingPeriodDataTableContents(): void {
       if (this.mapOfReportingPeriodToActiveDataset) {
-        let key: string;
-        let value: DataMetaInformation;
-        for ([key, value] of this.mapOfReportingPeriodToActiveDataset as Map<string, DataMetaInformation>) {
+        const sortedReportingPeriodMetaInfoPairs = Array.from(
+          (this.mapOfReportingPeriodToActiveDataset as Map<string, DataMetaInformation>).entries()
+        ).sort((firstEl, secondEl) => parseInt(firstEl[0]) - parseInt(secondEl[0]));
+        for (const [key, value] of sortedReportingPeriodMetaInfoPairs) {
```

Comment:
@fsoffel  I think you wrote this according to git blame.

This is just to make the the reporting periods be displayed in descending order (if they can be parsed as integers).

If a reporting Period cannot be parsed as integer, I think it will remain in the same position of the array, meaning that it will be just "ignored" in the sorting process.

Is this ok in your opinion?   I mean we could also handle the "NaN" case and do something with those (e.g. putting them at the end). 
I am not sure and don't have a strong opinion on that.

Reply:
Me neither. The reason for adding this was that we had a failing test, that relied on the smaller year number being the last element in the list in the edit button pop up. This was actually not always the case and that is what I was trying to fix here. I don't have any opinion on how to incorporate sorting of non-year reporting periods here. Do we know if we will have a fixed format for reporting periods (or do we have one already)? In that case we can implement proper sorting. Otherwise we could do as you suggested.

Reply:
We don't have a fixed format. Therefore I suggest we put all reportingPeriods that cannot be parsed as Integer simply at the end, and sort them alphabetically there.

We have done a similar thing when sorting the columns of the table on the Lksg-View-Page => The largest reportingPeriod number is always left, then descending, and at the very right you have everything that could not be parsed as Integer I think, sorted alphabetically.

The function is named `sortReportingPeriodsToDisplayAsColumns()`. You can use it as guidance.

Reply:
Ok, I will do that.

Reply:
Haben wir

Reply:
Nice thanks.

It is a little irritating that the methods names don't have verbs in.
Could you rename `reportingPeriodDataTableContents` to smth like `setReportingPeriodDataTableContents`
and `reportingPeriodComparator` to smth like `sortTwoReportingPeriods` ?

Reply:
I think the sorting is broken.
And why haven't you reused the sorting method which is already in place for the columns on the lksg view page?
The sorting in the EDIT dropdown should be the same, right?

Reply:
![image](https://github.com/d-fine/Dataland/assets/99255296/d35ebd9b-25d4-48a1-985f-b7d4aec71525)

Reply:
did the first exactly and set the second one to `compareReportingPeriods`

Reply:
The other sorting method did not consider some edge cases and had partly unexprected string comparison results ("12" < "122" < "13") and we agreed on the version I implemented with sorting everything that is not a year to the end. That was because we do not have a pattern specified. However I also did not want to change the existing sorting for the columns. Although consistency would be nice. What version would you rather prefer.

Reply:
Passt jetzt! Danke dir und schönes Wochenende 👍